pkg/network/network_test.go

```diff
@@ -0,0 +1,66 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Network", func() {
+	Describe("Internal Functions", func() {
+		It("should NOT return an error because kube mac pool can be nil", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should NOT return an error because both or none of the KubeMacPool can be configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should return an error because only start range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "02:00:00:00:00:00", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+		})
+
+		It("should return an error because only end range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: "02:00:00:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+
+		})
+		It("should return an error because the start range mac address is invalid", func() {
```

Comment:
Please describe what is invalid here (start > end, right?)

Reply:
yes, 'invalid' is a bit vague here; this specific address fails because it is too long (more than 6 octets). We should add another test for negative-length ranges.